### ISSUE TITLE
Rectangle drawing issues

### ISSUE THREAD

**Issue by bigpcz** [mono/xwt#374]
Strange garbage appears when drawing rectangles. It seems like the margin space is not redrawn correctly.

GTK backend @ windows:

![xwt-rectangledrawing_gtk_backend](https://cloud.githubusercontent.com/assets/7683458/3869596/49204be0-209b-11e4-9c38-adf138077c31.png)

WPF backend @ windows - doesn't have this issue:

![wpf](https://cloud.githubusercontent.com/assets/7683458/3869605/2a1d276c-209c-11e4-8be0-96e77715d485.png)

		protected override void OnDraw(Context ctx, Rectangle dirtyRect)
		{
			ctx.SetLineWidth(1);
			ctx.SetColor(Color.FromBytes(1, 1, 1));
			ctx.Rectangle(0, 0, this.Size.Width, this.Size.Height);
			ctx.Stroke();

			base.OnDraw(ctx, dirtyRect);
		}

**Comment by sevoku** [comment]
#431 should fix this (clips the drawing region)

**Comment by sevoku** [comment]
Ok, I've tested this again and #431 seems to have fixed this. Works on Windows and Linux (Mac should do it, too).
Test code:
```c#

	class MainClass
	{
		public static void Main (string[] args)
		{
			Application.Initialize (ToolkitType.Gtk);
			Window w = new Window ();
			w.Content = new Test ();
			w.Show ();
			Application.Run ();
			w.Dispose ();
			Application.Dispose ();
		}
	}

	class Test : Canvas
	{
		protected override void OnDraw(Context ctx, Rectangle dirtyRect)
		{
			ctx.SetLineWidth(1);
			ctx.SetColor(Color.FromBytes(1, 1, 1));
			ctx.Rectangle(0, 0, this.Size.Width, this.Size.Height);
			ctx.Stroke();

			base.OnDraw(ctx, dirtyRect);
		}
	}
```

(closed via #431)
